framework/LoopMeUnitedSDK/Core/Internal/Telemetry/VideoBufferingTracker.swift

```diff
@@ -0,0 +1,293 @@
+
+        item.removeObserver(self, forKeyPath: "playbackBufferEmpty", context: &playerItemContext)
+        log("Removed observer for 'playbackBufferEmpty'")
+
+        item.removeObserver(self, forKeyPath: "playbackLikelyToKeepUp", context: &playerItemContext)
+        log("Removed observer for 'playbackLikelyToKeepUp'")
+
+        // Remove the notification observer for this item
+        NotificationCenter.default.removeObserver(self, name: .AVPlayerItemDidPlayToEndTime, object: item)
+        log("Removed observer for AVPlayerItemDidPlayToEndTime on item")
+
+        currentPlayerItem = nil
+    }
+
+    /// Cancels all tracking and resets internal state.
+    @objc public func cancelTracking() {
+        log("Cancelling tracking and resetting state")
+        removeObservers()
+        totalBufferingDuration = 0
+        bufferingStartTime = nil
+        bufferCount = 0
+        isBuffering = false
+        mediaURL = nil
+    }
+
+    /// Handles the event when the player finishes playing an item.
+    /// - Parameter notification: The notification object.
+    @objc private func playerDidFinishPlaying(_ notification: Notification) {
+        log("Player did finish playing")
+        // Send the buffering event data
+        sendBufferingEvent()
+    }
+
+    /// Handles the event when the application enters the background.
+    /// - Parameter notification: The notification object.
+    @objc private func applicationDidEnterBackground(_ notification: Notification) {
+        log("Application did enter background")
+        // Send the buffering event data
+        sendBufferingEvent()
+    }
+
+    /// Called when the user skips the ad.
+    @objc public func userDidSkipAd() {
+        log("User did skip ad")
+        sendBufferingEvent()
+    }
+
+    /// Sends the collected buffering event data to the delegate.
+    private func sendBufferingEvent() {
+        log("Sending buffering event")
+
+        // Ensure buffering has ended
+        if isBuffering {
+            bufferingEnded()
+        }
+
+        // Calculate duration in seconds
+        let duration = totalBufferingDuration
+        // Calculate average duration per buffering event
+        let durationAvg = bufferCount > 0 ? Double(duration) / Double(bufferCount) : 0.0
+
+        // Only send event if total buffering duration is greater than 0
+        guard duration > 0, let mediaURL = mediaURL else {
+            log("sendBufferingEvent: No buffering duration or mediaURL available. Event not sent.")
+            return
+        }
+
+        let event = VideoBufferingEvent(
+            duration: NSNumber(value: duration),
+            durationAvg: NSNumber(value: durationAvg),
+            bufferCount: NSNumber(value: bufferCount),
+            mediaURL: mediaURL
+        )
+
+        log("Buffering Event - Duration: \(duration), Avg: \(durationAvg), Count: \(bufferCount), URL: \(mediaURL)")
+
+        delegate?.videoBufferingTracker(self, didCaptureEvent: event)
+
```

Comment:
@ltomaszewskiLoopme what do you think about add this delegate in main thread? Because it interacts with UI